File: app/src/main/java/com/marcelorcorrea/falae/adapter/SpreadSheetAdapter.java
package com.marcelorcorrea.falae.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.marcelorcorrea.falae.R;
import com.marcelorcorrea.falae.model.SpreadSheet;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by marcelo on 4/18/17.
 */

public class SpreadSheetAdapter extends RecyclerView.Adapter<SpreadSheetAdapter.SpreadSheetViewHolder> {
    public interface OnItemClickListener {
        void onItemClick(SpreadSheet spreadSheet);
    }

    private final SpreadSheetAdapter.OnItemClickListener listener;
    private List<SpreadSheet> spreadSheets;
    private Context context;

    public SpreadSheetAdapter(Context context, List<SpreadSheet> spreadSheets, SpreadSheetAdapter.OnItemClickListener listener) {
        this.spreadSheets = spreadSheets;
        this.context = context;
        this.listener = listener;
    }

    public void update(List<SpreadSheet> spreadSheets) {
        this.spreadSheets = spreadSheets;
        notifyDataSetChanged();
    }

    @Override
    public SpreadSheetViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.spreadsheet_item, parent, false);
        return new SpreadSheetViewHolder(view);
    }

    @Override
    public void onBindViewHolder(SpreadSheetViewHolder holder, int position) {
        holder.bind(spreadSheets.get(position), listener);
    }

    @Override
    public int getItemCount() {
        return spreadSheets.size();
    }

    public class SpreadSheetViewHolder extends RecyclerView.ViewHolder {
        final TextView name;

        SpreadSheetViewHolder(View view) {
            super(view);
            name = (TextView) view.findViewById(R.id.spreadsheet_name);
        }

        void bind(final SpreadSheet spreadSheet, final SpreadSheetAdapter.OnItemClickListener onItemClickListener) {
            name.setText(spreadSheet.getName());
            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    onItemClickListener.onItemClick(spreadSheet);
                }
            });

        }
    }
}

File: app/src/main/java/com/marcelorcorrea/falae/adapter/SpreadSheetAdapter.kt
package com.marcelorcorrea.falae.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.marcelorcorrea.falae.R
import com.marcelorcorrea.falae.model.SpreadSheet

/**
 * Created by marcelo on 4/18/17.
 */

class SpreadSheetAdapter(private val context: Context, private val spreadSheets: List<SpreadSheet>, private val onItemClick: (spreadSheet: SpreadSheet) -> Unit) : RecyclerView.Adapter<SpreadSheetAdapter.SpreadSheetViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SpreadSheetViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.spreadsheet_item, parent, false)
        return SpreadSheetViewHolder(view)
    }

    override fun onBindViewHolder(holder: SpreadSheetViewHolder, position: Int) {
        holder.bind(spreadSheets[position], onItemClick)
    }

    override fun getItemCount(): Int = spreadSheets.size

    inner class SpreadSheetViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        internal val name: TextView = view.findViewById(R.id.spreadsheet_name) as TextView

        internal fun bind(spreadSheet: SpreadSheet, onItemClickListener: (spreadSheet: SpreadSheet) -> Unit) {
            name.text = spreadSheet.name
            itemView.setOnClickListener { onItemClickListener(spreadSheet) }

        }
    }
}
